File: benchmark/prime_torture.py
import sys

NUM = 750000

def is_prime(n: int) -> bool:
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            return False
    return True

def prime_torture(n: int):
    for i in range(1, n+1):
        if is_prime(i):
            sys.stdout.write(str(i) + ", ")
            sys.stdout.flush()

prime_torture(NUM)
print()

File: benchmark/prime_torture.c
#include <math.h>
#include <stdio.h>

#define NUM 750000

int is_prime(int n) {
    for (int i = 2; i <= (int)sqrt(n); i++) {
        if (n % i == 0)
            return 0;
    }

    return 1;
}

void prime_torture(int n) {
    for (int i = 1; i <= n; i++) {
        if (is_prime(i))
            printf("%d, ", i);
        fflush(stdout);
    }
}

int main(void) {
    prime_torture(NUM);
    puts("");
    return 0;
}
